Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -1302,6 +1303,54 @@ HTTPServer server({
     EXPECT_WK_STREQ("mousedown,40,40", eventTypes[1]);
     EXPECT_WK_STREQ("mouseup,40,40", eventTypes[2]);
 }
+
+TEST(SiteIsolation, DragEvents)
+{
+    auto mainframeHTML = "<script>"
+    "    window.events = [];"
+    "    addEventListener('message', function(event) {"
+    "        window.events.push(event.data);"
+    "    });"
+    "</script>"
+    "<iframe width='300' height='300' src='https://domain2.com/subframe'></iframe>"_s;
+
+    auto subframeHTML = "<body>"
+    "<div id='draggable' draggable='true' style='width: 100px; height: 100px; background-color: blue;'></div>"
+    "<script>"
+    "    draggable.addEventListener('dragstart', window.parent.postMessage('dragstart', '*'));"
+    "    draggable.addEventListener('drag', window.parent.postMessage('drag', '*'));"
+    "    draggable.addEventListener('dragend', window.parent.postMessage('dragend', '*'));"
+    "    draggable.addEventListener('dragenter', window.parent.postMessage('dragenter', '*'));"
+    "    draggable.addEventListener('dragleave', window.parent.postMessage('dragleave', '*'));"
+    "</script>"
+    "</body>"_s;
+
+    HTTPServer server({
+        { "/mainframe"_s, { mainframeHTML } },
+        { "/subframe"_s, { subframeHTML } }
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
+    [navigationDelegate allowAnyTLSCertificate];
+    auto configuration = server.httpsProxyConfiguration();
+    enableSiteIsolation(configuration);
+    auto simulator = adoptNS([[DragAndDropSimulator alloc] initWithWebViewFrame:NSMakeRect(0, 0, 400, 400) configuration:configuration]);
+    auto webView = adoptNS([simulator webView]);
```

Comment:
This is an overrelease, since `[simulator webView]` doesn't return a +1 object. We can just do this instead:
```suggestion
    RetainPtr webView = [simulator webView];
```